app/controllers/media/objects_controller.rb

```diff
@@ -29,4 +45,17 @@ def media_filename
     extension = File.extname(@media_object.item_file_name)
     "elmo-#{@response.id}-#{@answer.id}#{extension}"
   end
+
+  def media_class(type)
+    case type
+    when 'audio'
+      return Media::Audio
+    when 'video'
+      return Media::Video
+    when 'image'
+      return Media::Image
+    else
+      return Media::Object
```

Comment:
It shouldn't ever happen, the `type` comes from the URL params and the scope restricts to the valid types, so that condition should be more-or-less unreachable entirely. Raising an error there wouldn't be a bad idea.